docs/contributing.md

```diff
@@ -0,0 +1,304 @@
+and formatting are provided automatically when you're editing a frontend file.
+
+#### Backend linting/formatting
+
+All of our python related components of elpiscloud use [black](https://github.com/psf/black)
+for formatting/linting.
+
+Within a poetry folder (`functions` or `services/**`), after you've installed  
+the dependencies, you can run `poetry run black .` to format the directory.
+
+This can also be done through vscode extensions, but whatever floats your boat.
+
+### Testing your code
+
+Before you start the process of submitting a PR, it is important to write some
+tests for your feature (if applicable).
+
+For our backend, we're using `pytest` and `pytest-mock` for testing.
+On the frontend, we don't currently have tests, but this may change in the future.
+
+#### Testing poetry environments
+
+From within a poetry folder (`functions` or `services/**`), you can run the
+full test suite with `poetry run pytest`.
+
+For the services, this isn't always recommended since in the integration tests,
+we perform a full training/inference run through, which can take a while.
+
+To run the non-integration tests from within a service, you can instead run:
+
+`poetry run pytest -m 'not integration'`
+
+Which will skip those tests.
+
+#### Writing your own tests
+
+When writing your own tests, I'd recommend following the style of the preexisting
+tests for that folder. Generally in poetry folders, the tests exist under the
+`tests` directory.
+
+As a general naming convention for tests you write, we recommend:
+`test_some_function_with_x_should_do_y()`.
+
+e.g.:
+
+- `test_double`
+- `test_process_dataset_with_invalid_data_should_raise_exception`
+- `test_extract_annotations_with_elan_file_should_call_extract_elan`
+
+For the **positive** case where the function performs as expected, you can omit the
+_with_ and _should_ sections of the name.
+
+We don't expect for you to write docstrings for any of the test functions you
+write, but as with the source code, if you're making an assertion about some
+expected value that might not be immediately clear in the code, there's no harm
+in adding a comment.
+
+### Commit your changes
+
+When you're happy with the code on your computer, you need to commit your changes:
+
+`git commit -a`
+
+This should fire up your editor to write a commit message. When you have finished,
+save, and close to continue.
+
+**TODO** Commit format
+
+### Update your branch
+
+It's likely that other changes to `main` may have happened while you were working.
+To update your branch with these changes:
+
+```
+git checkout main
+git pull --rebase
+```
+
+Now apply those changes to your branch:
+
+```
+git rebase my_new_branch
+git checkout my_new_branch
+```
+
+If there are no conflicts, and after running the tests, all seems well at the inn,
+its time to proceed.
+
+### Forking (general contributors)
+
+First, navigate to the [elpiscloud github repo](https://github.com/CoEDL/elpiscloud)
```

Comment:
Might be worth explaining briefly why forking is necessary for non-org contributors.

Reply:
Or leading with the CoEDL section.

Reply:
Fixed in latest revision.